### PR TITLE
DSD-548: Horizontal Rule - height prop validation

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-548](https://jira.nypl.org/browse/DSD-548)
[DSD-558](https://jira.nypl.org/browse/DSD-558)

## This PR does the following:
- Adds validation to the `height` prop in the `HorizontalRule`.
- Also updates the latest version to the correct version in the `List` storybook doc.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/HorizontalRule/HorizontalRule.tsx [hunk 1]
```diff
@@ -3,13 +3,19 @@ import * as React from "react";
...
+  /** Optional alignment value to align the horizontal rule to one side or the
+   * other when the width is less than 100%. If omitted, the horizontal rule
+   * will have a default center alignment. */
   align?: "left" | "right";
   /** ClassName you can add in addition to `horizontal-rule` */
   className?: string;
-  /** Optional height value.  This value should be entered with the same formatting as a CSS height attribute (ex. `2%`, `5px`, `1.5rem`).  If omitted, the horizobntal rule will have a default height of 2px. */
+  /** Optional height value. This value should be entered with the same
+   * formatting as a CSS height attribute (ex. `2`, `5px`, `1.5rem`). If
+   * omitted, the horizontal rule will have a default height of 2px. */
```
Comment (bigfishdesign13) [inline src/components/HorizontalRule/HorizontalRule.tsx]: Add somethign here about percent values not being valid...
```If omitted or if a percent value is used (ex. `20%`), the...```

File: src/components/HorizontalRule/HorizontalRule.tsx [hunk 2]
```diff
@@ -18,10 +24,19 @@ export default function HorizontalRule(
 ) {
   const { align, className, height = "2px", width = "auto" } = props;
   const styles = useStyleConfig("HorizontalRule", { align });
+  let finalHeight = height;
+
+  if (height.endsWith("%")) {
+    console.warn(
+      "For the `height` prop, use a whole number, a `px` value, a `em` value, or a `rem` value. Using the default of 2px."
```
Comment (bigfishdesign13) [inline src/components/HorizontalRule/HorizontalRule.tsx]: I don't think you need the last bit.  I also would clarify where this warning is coming from.
```
For the HorizontalRule `height` prop, a whole number, a `px` value, a `em` value, or a `rem` value mus be used.
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
A few language changes needed.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr751-dl3x14dvnoqebn7tqojhzbyj8ygn2aob.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61828f4c908ae18fd050b309

**Review by bigfishdesign13 — APPROVED** [review]
Magnificent.

**Pull request merged**